x-pack/plugins/infra/public/observability_logs/log_stream_query_state/src/url_state_storage_service.ts

```diff
@@ -10,28 +10,53 @@ import { Query } from '@kbn/es-query';
 import { IKbnUrlStateStorage, withNotifyOnErrors } from '@kbn/kibana-utils-plugin/public';
 import * as Array from 'fp-ts/lib/Array';
 import * as Either from 'fp-ts/lib/Either';
-import { pipe } from 'fp-ts/lib/function';
+import { identity, pipe } from 'fp-ts/lib/function';
 import * as rt from 'io-ts';
 import { InvokeCreator } from 'xstate';
+import { DurationInputObject } from 'moment';
+import moment from 'moment';
+import { minimalTimeKeyRT } from '../../../../common/time';
+import { datemathStringRT } from '../../../utils/datemath';
 import { createPlainError, formatErrors } from '../../../../common/runtime_types';
 import { replaceStateKeyInQueryString } from '../../../utils/url_state';
 import type { LogStreamQueryContext, LogStreamQueryEvent, ParsedQuery } from './types';
 
 interface LogStreamQueryUrlStateDependencies {
   filterStateKey?: string;
+  positionStateKey?: string;
   savedQueryIdKey?: string;
   toastsService: IToasts;
   urlStateStorage: IKbnUrlStateStorage;
 }
 
 const defaultFilterStateKey = 'logFilter';
-const defaultFilterStateValue: Required<FilterStateInUrl> = {
+const defaultPositionStateKey = 'logPosition'; // NOTE: Provides backwards compatibility for start / end / streamLive previously stored under the logPosition key.
+
+type RequiredDefaults = Required<Omit<FilterStateInUrl, 'timeRange' | 'refreshInterval'>>;
+type OptionalDefaults = Pick<FilterStateInUrl, 'timeRange' | 'refreshInterval'>;
+type FullDefaults = Required<RequiredDefaults & OptionalDefaults>;
+
+const requiredDefaultFilterStateValue: RequiredDefaults = {
   query: {
     language: 'kuery',
     query: '',
   },
   filters: [],
 };
+
+const optionalDefaultFilterStateValue = {
+  timeRange: {
+    from: 'now-1d',
+    to: 'now',
+  },
+  refreshInterval: { pause: true, value: 5000 },
+};
```

Comment:
**question:** I have seen these default values before in this PR, is it something that should be consistent between this service and the same definitions inside the state machine and other services? Shall we abstract this into a configuration?

Reply:
Ah good catch. I actually meant to revisit this, to tidy it up. I can probably just remove these from here (since we go defaults > time filter service > URL and if these values are not defined in the URL that's fine).